BUILD.sh

```diff
@@ -1,69 +1,226 @@
+  echo
+  echo Exit code $1 caught in build script.
+  echo
+  if [[ "$1" == "0" ]]; then
+    echo BUILD_STATUS:0
+  else
+    echo
+    echo Failure found on line $2 in build script.
+    echo
+    echo BUILD_STATUS:1
+  fi
+}
 
-#  NOTE: The following is required when running from Gitlab CI via slurm job
-source /etc/profile.d/modules.sh
+trap 'cleanup $? $LINENO' EXIT
 
-export MY_CLUSTER=`uname -n | sed -e 's/[0-9]//g' -e 's/\..*//'`
-export NVBLAS_CONFIG_FILE=$(pwd)/nvblas.conf
+export BUILD=1
+export TEST=1
+while [[ $# -gt 0 ]]
+do
+  case $1 in
+    --build-only|-B)
+      echo
+      echo Building only
+      echo
+      export BUILD=1
+      export TEST=0
+      shift
+      ;;
+    --test-only|-T)
+      echo
+      echo Testing only
+      echo
+      export BUILD=0
+      export TEST=1
+      shift
+      ;;
+    *)
+      cat <<EOD
+    Argument not found!
+      
+    usage: $0 [ --test-only|-T ] [ --build-only|-B ]
+EOD
+      exit 1
+      ;;
+  esac
+done
+
+set -x
+x="unset"
+
+if [[ ! -v MY_CLUSTER ]]
+then
+  export MY_CLUSTER=`uname -n | sed -e 's/[0-9]//g' -e 's/\..*//'`
+fi
+BUILDDIR="$(pwd)/build"
+extra_cmake_args=""
 
 module purge
-if [ "$MY_CLUSTER" == "newell" ]; then
+case "$MY_CLUSTER" in
+newell|newell_shared)
+    export PROJ_DIR=/qfs/projects/exasgd
+    export APPS_DIR=/share/apps
+    #  NOTE: The following is required when running from Gitlab CI via slurm job
+    source /etc/profile.d/modules.sh
+    module use -a /usr/share/Modules/modulefiles
+    module use -a /share/apps/modules/tools
+    module use -a /share/apps/modules/compilers
+    module use -a /share/apps/modules/mpi
+    module use -a /etc/modulefiles
     export MY_GCC_VERSION=7.4.0
     export MY_CUDA_VERSION=10.2
     export MY_OPENMPI_VERSION=3.1.5
     export MY_CMAKE_VERSION=3.16.4
     export MY_MAGMA_VERSION=2.5.2_cuda10.2
-    export MY_HIOP_MAGMA_DIR=/share/apps/magma/2.5.2/cuda10.2
+    export MY_METIS_VERSION=5.1.0
+    export MY_NVCC_ARCH="sm_70"
+
+    module load gcc/$MY_GCC_VERSION
+    module load cuda/$MY_CUDA_VERSION
+    module load openmpi/$MY_OPENMPI_VERSION
+    module load cmake/$MY_CMAKE_VERSION
     module load magma/$MY_MAGMA_VERSION
+
+    export NVBLAS_CONFIG_FILE=$PROJ_DIR/$MY_CLUSTER/nvblas.conf
+    export MY_RAJA_DIR=$PROJ_DIR/$MY_CLUSTER/raja
+    export MY_UMPIRE_DIR=$PROJ_DIR/$MY_CLUSTER/umpire
+    export MY_UMFPACK_DIR=$PROJ_DIR/$MY_CLUSTER/suitesparse
+    export MY_METIS_DIR=$APPS_DIR/metis/$MY_METIS_VERSION
+    export MY_HIOP_MAGMA_DIR=$APPS_DIR/magma/2.5.2/cuda10.2
+    ;;
+ascent)
+    module purge
+    module load cuda/11
+    module use /gpfs/wolf/proj-shared/csc359/ascent/Modulefiles/Core
+    module load exasgd-base
+    module load gcc-ext/7.4.0
+    module load spectrum-mpi-ext
+    module load openblas
+    module load magma/2.5.3-cuda11
+    module load metis
+    module load mpfr
+    module load suitesparse
+    module load cmake/3.18.2
+    module load raja
+    module load umpire
```

Comment:
Tried to run these commands in login1.ascent, but many modules are missing. For example, metis.
Should they be provided by `/gpfs/wolf/proj-shared/csc359/ascent/Modulefiles/Core`?

Reply:
Metis is found above on line 103. What other modules seem to be missing?

Reply:
FYI there seems to be an issue with metis installation on Ascent. We have a separate issue to track this on ORNL gitlab if you have an account there.